Compilation issue on Windows

Real C++ programmer compile library from source. So I clone the project then do a 
```powershell
# blahblahblah env setup
# ...
python setup.py develop
# or
python setup.py build_ext
```

# The first issue is
```
...
CMake Error at chainerx/native/CMakeLists.txt:46 (list):
  list GET given empty list


CMake Error at chainerx/native/CMakeLists.txt:47 (get_filename_component):
  get_filename_component called with incorrect number of arguments


CMake Error at chainerx/native/CMakeLists.txt:48 (get_filename_component):
  get_filename_component called with incorrect number of arguments

...
```

# The second issue is
<details><summary>Non position independent code link error</summary>
<p>

```
...
gtest.lib(gtest-all.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_streambuf<char,struct std::char_traits<c
har> >::~basic_streambuf<char,struct std::char_traits<char> >(void)" (??1?$basic_streambuf@DU?$char_traits@D@std@@@std@
@UEAA@XZ) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python
\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_ios<char,struct std::char_traits<char> >
::~basic_ios<char,struct std::char_traits<char> >(void)" (??1?$basic_ios@DU?$char_traits@D@std@@@std@@UEAA@XZ) already
defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj
]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: virtual __cdecl std::basic_ostream<char,struct std::char_traits<cha
r> >::~basic_ostream<char,struct std::char_traits<char> >(void)" (??1?$basic_ostream@DU?$char_traits@D@std@@@std@@UEAA@
XZ) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core
.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: void __cdecl std::basic_ostream<char,struct std::char_traits<char>
>::_Osfx(void)" (?_Osfx@?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAAXXZ) already defined in gtest.lib(gtest-all.ob
j) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(int)" (??6?$basic_ostream@DU?$char_traits@D@st
d@@@std@@QEAAAEAV01@H@Z) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Release\
chainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(__int64)" (??6?$basic_ostream@DU?$char_traits@
D@std@@@std@@QEAAAEAV01@_J@Z) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Rel
ease\chainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::flush(void)" (?flush@?$basic_ostream@DU?$char_traits@D@st
d@@@std@@QEAAAEAV12@XZ) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Release\c
hainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(unsigned __int64)" (??6?$basic_ostream@DU?$cha
r_traits@D@std@@@std@@QEAAAEAV01@_K@Z) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd6
4-3.7\Release\chainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(double)" (??6?$basic_ostream@DU?$char_traits@D
@std@@@std@@QEAAAEAV01@N@Z) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Relea
se\chainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: char __cdecl std::basic_ios<char,struct std::char_traits<char> >::w
iden(char)const " (?widen@?$basic_ios@DU?$char_traits@D@std@@@std@@QEBADD@Z) already defined in gtest.lib(gtest-all.obj
) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::put(char)" (?put@?$basic_ostream@DU?$char_traits@D@std@@@
std@@QEAAAEAV12@D@Z) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chai
nerx\python\_core.pyd.vcxproj]
msvcprt.lib(MSVCP140.dll) : error LNK2005: "public: class std::basic_ostream<char,struct std::char_traits<char> > & __c
decl std::basic_ostream<char,struct std::char_traits<char> >::operator<<(void const *)" (??6?$basic_ostream@DU?$char_tr
aits@D@std@@@std@@QEAAAEAV01@PEBX@Z) already defined in gtest.lib(gtest-all.obj) [D:\test\chainer\build\temp.win-amd64-
3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale0.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
libcpmt.lib(locale0.obj) : error LNK2005: "void __cdecl std::_Facet_Register(class std::_Facet_base *)" (?_Facet_Regist
er@std@@YAXPEAV_Facet_base@1@@Z) already defined in msvcprt.lib(locale0_implib.obj) [D:\test\chainer\build\temp.win-amd
64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale0.obj) : error LNK2005: "private: static class std::locale::_Locimp * __cdecl std::locale::_Getglobal
locale(void)" (?_Getgloballocale@locale@std@@CAPEAV_Locimp@12@XZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test
\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale0.obj) : error LNK2005: "private: static class std::locale::_Locimp * __cdecl std::locale::_Init(bool
)" (?_Init@locale@std@@CAPEAV_Locimp@12@_N@Z) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.
win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale0.obj) : error LNK2005: "public: static void __cdecl std::_Locinfo::_Locinfo_ctor(class std::_Locinfo
 *,char const *)" (?_Locinfo_ctor@_Locinfo@std@@SAXPEAV12@PEBD@Z) already defined in msvcprt.lib(MSVCP140.dll) [D:\test
\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale0.obj) : error LNK2005: "public: static void __cdecl std::_Locinfo::_Locinfo_dtor(class std::_Locinfo
 *)" (?_Locinfo_dtor@_Locinfo@std@@SAXPEAV12@@Z) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\te
mp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(cerr.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match v
alue 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd
.vcxproj]
libcpmt.lib(cerr.obj) : error LNK2005: "protected: __cdecl std::basic_ios<char,struct std::char_traits<char> >::basic_i
os<char,struct std::char_traits<char> >(void)" (??0?$basic_ios@DU?$char_traits@D@std@@@std@@IEAA@XZ) already defined in
 msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(cerr.obj) : error LNK2005: "public: __cdecl std::basic_ostream<char,struct std::char_traits<char> >::basic_
ostream<char,struct std::char_traits<char> >(class std::basic_streambuf<char,struct std::char_traits<char> > *,bool)" (
??0?$basic_ostream@DU?$char_traits@D@std@@@std@@QEAA@PEAV?$basic_streambuf@DU?$char_traits@D@std@@@1@_N@Z) already defi
ned in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(cerr.obj) : error LNK2005: "protected: __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::b
asic_streambuf<char,struct std::char_traits<char> >(void)" (??0?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAA@XZ)
already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.py
d.vcxproj]
libcpmt.lib(cerr.obj) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<cha
r> >::_Pninc(void)" (?_Pninc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAPEADXZ) already defined in msvcprt.lib(
MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(cerr.obj) : error LNK2005: "public: void __cdecl std::basic_ios<char,struct std::char_traits<char> >::setst
ate(int,bool)" (?setstate@?$basic_ios@DU?$char_traits@D@std@@@std@@QEAAXH_N@Z) already defined in msvcprt.lib(MSVCP140.
dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match
 value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.p
yd.vcxproj]
libcpmt.lib(locale.obj) : error LNK2005: "public: struct _Cvtvec __cdecl std::_Locinfo::_Getcvt(void)const " (?_Getcvt@
_Locinfo@std@@QEBA?AU_Cvtvec@@XZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.
7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale.obj) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<c
har> >::_Pninc(void)" (?_Pninc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAPEADXZ) already defined in msvcprt.li
b(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(locale.obj) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char> >
::sputc(char)" (?sputc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHD@Z) already defined in msvcprt.lib(MSVCP140
.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(iosptrs.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
libcpmt.lib(xlock.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match
value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.py
d.vcxproj]
libcpmt.lib(xlock.obj) : error LNK2005: "public: __cdecl std::_Lockit::_Lockit(int)" (??0_Lockit@std@@QEAA@H@Z) already
 defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxpr
oj]
libcpmt.lib(xlock.obj) : error LNK2005: "public: __cdecl std::_Lockit::~_Lockit(void)" (??1_Lockit@std@@QEAA@XZ) alread
y defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxp
roj]
libcpmt.lib(wlocale.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
libcpmt.lib(wlocale.obj) : error LNK2005: "public: struct _Cvtvec __cdecl std::_Locinfo::_Getcvt(void)const " (?_Getcvt
@_Locinfo@std@@QEBA?AU_Cvtvec@@XZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3
.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(wlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getdays(void)const
" (?_W_Getdays@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-a
md64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(wlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getmonths(void)cons
t " (?_W_Getmonths@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.w
in-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xlocale.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
libcpmt.lib(xlocale.obj) : error LNK2005: "public: struct _Cvtvec __cdecl std::_Locinfo::_Getcvt(void)const " (?_Getcvt
@_Locinfo@std@@QEBA?AU_Cvtvec@@XZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3
.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xlocale.obj) : error LNK2005: "protected: char * __cdecl std::basic_streambuf<char,struct std::char_traits<
char> >::_Pninc(void)" (?_Pninc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAPEADXZ) already defined in msvcprt.l
ib(MSVCP140.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getdays(void)const
" (?_W_Getdays@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.win-a
md64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xlocale.obj) : error LNK2005: "public: unsigned short const * __cdecl std::_Locinfo::_W_Getmonths(void)cons
t " (?_W_Getmonths@_Locinfo@std@@QEBAPEBGXZ) already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\temp.w
in-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xlocale.obj) : error LNK2005: "public: int __cdecl std::basic_streambuf<char,struct std::char_traits<char>
>::sputc(char)" (?sputc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAHD@Z) already defined in msvcprt.lib(MSVCP14
0.dll) [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xstol.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match
value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.py
d.vcxproj]
libcpmt.lib(xstoul.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match
 value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.p
yd.vcxproj]
libcpmt.lib(xstoll.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match
 value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.p
yd.vcxproj]
libcpmt.lib(xstoull.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
libcpmt.lib(xmtx.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match v
alue 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd
.vcxproj]
libcpmt.lib(xstrcoll.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't mat
ch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core
.pyd.vcxproj]
libcpmt.lib(xdateord.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't mat
ch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core
.pyd.vcxproj]
libcpmt.lib(xwctomb.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't matc
h value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.
pyd.vcxproj]
libcpmt.lib(xwctomb.obj) : error LNK2005: _Getcvt already defined in msvcprt.lib(MSVCP140.dll) [D:\test\chainer\build\t
emp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
libcpmt.lib(xwcscoll.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't mat
ch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core
.pyd.vcxproj]
libcpmt.lib(xwcsxfrm.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't mat
ch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core
.pyd.vcxproj]
libcpmt.lib(xgetwctype.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't m
atch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_co
re.pyd.vcxproj]
libcpmt.lib(xtowlower.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't ma
tch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_cor
e.pyd.vcxproj]
libcpmt.lib(xtowupper.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't ma
tch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_cor
e.pyd.vcxproj]
libcpmt.lib(xstrxfrm.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't mat
ch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core
.pyd.vcxproj]
libcpmt.lib(winapisupp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't m
atch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_co
re.pyd.vcxproj]
libcpmt.lib(StlCompareStringA.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' do
esn't match value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\pyt
hon\_core.pyd.vcxproj]
libcpmt.lib(winapinls.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't ma
tch value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_cor
e.pyd.vcxproj]
libcpmt.lib(StlCompareStringW.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' do
esn't match value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\pyt
hon\_core.pyd.vcxproj]
libcpmt.lib(StlLCMapStringW.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' does
n't match value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\pytho
n\_core.pyd.vcxproj]
libcpmt.lib(StlLCMapStringA.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' does
n't match value 'MD_DynamicRelease' in core_module.obj [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\pytho
n\_core.pyd.vcxproj]
     Creating library D:/test/chainer/build/temp.win-amd64-3.7/Release/chainerx/python/Release/_core.pyd.lib and object
   D:/test/chainer/build/temp.win-amd64-3.7/Release/chainerx/python/Release/_core.pyd.exp
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\test\chaine
r\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
D:\test\chainer\build\lib.win-amd64-3.7\chainerx\_core.pyd : fatal error LNK1169: one or more multiply defined symbols
found [D:\test\chainer\build\temp.win-amd64-3.7\Release\chainerx\python\_core.pyd.vcxproj]
Traceback (most recent call last):
  File ".\setup.py", line 198, in <module>
    setup(**setup_kwargs)
  File "C:\Users\cloud\Miniconda3\lib\site-packages\setuptools\__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "C:\Users\cloud\Miniconda3\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Users\cloud\Miniconda3\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Users\cloud\Miniconda3\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "D:\test\chainer\chainerx_build_helper.py", line 44, in run
    self.build_extension(ext)
  File "D:\test\chainer\chainerx_build_helper.py", line 95, in build_extension
    ['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
  File "C:\Users\cloud\Miniconda3\lib\subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '/m']' returned non-zero exit status 1.
```
</p>
</details>

there are may other minor issues, which generate annoying warnings.
1. `warning C4244:  'return': conversion from 'double' to 'int32_t', possible loss of data`
2. nvcc propagate ill-formed `-std=c++14` commandline switch to host compiler. In this case, the host compiler `cl` accept `/std:c++14` is IIRC

--- a/chainerx_cc/chainerx/native/CMakeLists.txt
+++ b/chainerx_cc/chainerx/native/CMakeLists.txt
@@ -40,7 +40,7 @@ add_library(chainerx_native STATIC
 if(${BLAS_FOUND})
     if(DEFINED ENV{CHAINERX_BLAS_INCLUDE_DIRS})
         set(BLAS_INCLUDE_DIRS $ENV{CHAINERX_BLAS_INCLUDE_DIRS})
-    else()
+    elseif(UNIX)
         # Workaround for UNIX environments to guess the BLAS include directory from its library path (.so).
         # This workaround is required since CMake does not set the BLAS_INCLUDE_DIRS variable.
         list(GET BLAS_LIBRARIES 0 BLAS_LIBRARY)

